From 6e2a53afa15422ee290663dbb798c085ef7068ed Mon Sep 17 00:00:00 2001
Message-Id: <6e2a53afa15422ee290663dbb798c085ef7068ed.1607538402.git.edvin.torok@citrix.com>
From: Wei Liu <wei.liu2@citrix.com>
Date: Mon, 23 Jul 2018 11:26:49 +0100
Subject: ocaml: remove undefined behaviour in systemd_stubs.c

Clang complains:

systemd_stubs.c:51:8: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
        ret = Val_int(-1U);
              ^~~~~~~~~~~~

Since sd_notify_fd has a signature of unit -> unit, we simply change
the return value to Val_unit.

Signed-off-by: Wei Liu <wei.liu2@citrix.com>
Acked-by: Christian Lindig <christian.lindig@citrix.com>
Acked-by: Andrew Cooper <andrew.cooper3@citrix.com>

diff --git a/tools/ocaml/xenstored/systemd_stubs.c b/tools/ocaml/xenstored/systemd_stubs.c
index a78a72b75c..046666f209 100644
--- a/tools/ocaml/xenstored/systemd_stubs.c
+++ b/tools/ocaml/xenstored/systemd_stubs.c
@@ -108,13 +108,10 @@ CAMLprim value ocaml_launched_by_systemd(value ignore)
 CAMLprim value ocaml_sd_notify_ready(value ignore)
 {
 	CAMLparam1(ignore);
-	CAMLlocal1(ret);
-
-	ret = Val_int(0);
 
 	sd_notify(1, "READY=1");
 
-	CAMLreturn(ret);
+	CAMLreturn(Val_unit);
 }
 
 #else
@@ -142,10 +139,7 @@ CAMLprim value ocaml_launched_by_systemd(value ignore)
 CAMLprim value ocaml_sd_notify_ready(value ignore)
 {
 	CAMLparam1(ignore);
-	CAMLlocal1(ret);
-
-	ret = Val_int(-1U);
 
-	CAMLreturn(ret);
+	CAMLreturn(Val_unit);
 }
 #endif
